# Comparative Analysis of Descriptive Metadata

### Newcastle University Special Collections and University of Edinburgh Archives

***

**Table of Contents**
* [Description Lengths](#description-lengths)
* [Gendered Language](#gendered-language)
* [Parts of Speech](#parts-of-speech)
  * [Adjectives](#adjectives)
  * [Adverbs](#adverbs)

***

In [1]:
import pandas as pd
import numpy as np

Create variables to reference the files and data points from the `analysis_metadata_XXX.ipynb` notebooks.

In [2]:
dir_nusc = "data/analysis/"
files_nusc = [
    "nusc_ead_descs_stats.csv", 
    "nusc_ead_gendered_lower_token_counts_ead.csv", "nusc_ead_gendered_capitalized_token_counts.csv",
    "nusc_ead_descs_pos_tags.csv",
    "nusc_ead_nltk_adj_token_counts.csv", "nusc_ead_nltk_adj_token_lower_counts.csv",
    "nusc_ead_nltk_adv_token_counts.csv", "nusc_ead_nltk_adv_token_lower_counts.csv",
    "nusc_ead_nltk_adj_token_counts.csv", "nusc_ead_nltk_adj_adv_stats.csv"
]
nusc_descs_total = 75165
nusc_sentences_total = 124592
nusc_tokens_total = 1915370    # all tokens, with repeats
nusc_vocab_size = 42531        # unique tokens, no repeats

In [3]:
dir_uoe = "data/uoe/analysis/"
files_uoe = [
    "uoe_ead_descs-oct2020_stats.csv", 
    "uoe_ead_gendered_lower_token_counts_oct2020.csv", "uoe_ead_gendered_capitalized_token_counts_oct2020.csv",
    "uoe_ead_descs_oct2020_pos_tags.csv", 
    "uoe_descs_oct2020_nltk_adj_token_counts.csv", "uoe_descs_oct2020_nltk_adj_token_lower_counts.csv",
    "uoe_descs_oct2020_nltk_adv_token_counts.csv", "uoe_descs_oct2020_nltk_adv_token_lower_counts.csv",
    "uoe_descs_oct2020_nltk_adj_adv_counts.csv", "uoe_oct2020_nltk_adj_adv_stats.csv"
]
uoe_descs_total = 27752
uoe_sentences_total = 41671
uoe_tokens_total = 542635     # all tokens, with repeats
uoe_vocab_size = 40949        # unique tokens, no repeats

## Description Lengths

In [ ]:
pd.set_option('display.float_format', lambda x: '%.3f' % x)  # display floating point numbers with 3 decimal places

In [7]:
df_len_nusc = pd.read_csv(dir_nusc + "nusc_ead_descs_stats.csv", index_col=0)
df_len_nusc

sentences_per_description  tokens_per_description  \
mean                                    1.658                  25.482   
median                                  1.000                   9.000   
standard_deviation                      2.396                  63.481   
minimum                                 1.000                   0.000   
maximum                                91.000                2296.000   
total                              124592.000             1915370.000   

                    tokens_per_sentence  
mean                             17.538  
median                           13.000  
standard_deviation               16.256  
minimum                           1.000  
maximum                        1676.000  
total                       1915370.000

In [8]:
df_len_uoe = pd.read_csv(dir_uoe + "uoe_ead_descs_oct2020_stats.csv", index_col=0)
df_len_uoe

sentences_per_description  tokens_per_description  \
mean                                    1.502                  19.553   
median                                  1.000                  10.000   
standard_deviation                      5.377                  97.627   
minimum                                 1.000                   1.000   
maximum                               742.000               14097.000   
total                               41671.000              542635.000   

                    tokens_per_sentence  
mean                              1.502  
median                            1.000  
standard_deviation                5.377  
minimum                           1.000  
maximum                         742.000  
total                        542635.000

Edinburgh's data shows more tokens per description on average, which I hadn't expected, though the median is greater for Edinburgh (10 vs. 9).  There is also greater variation in description lengths in the Edinburgh data relative to the Newcastle data.  Also, the maximum number of tokens in a single description is far larger (14,097 vs. 2,296), as is the maximum number of sentences (742 vs. 91). 

TOKENS PER SENTENCE FOR UOE DOESN'T LOOK RIGHT!

In [ ]:
mean_tokens_per_desc = list(df_len_uoe["tokens_per_description"])[0]